csharp/src/Google.Protobuf.Test/FieldMaskTreeTest.cs

```diff
@@ -380,6 +380,20 @@ public void Merge(bool useDynamicMessage)
                 clearedSource, destination, options, useDynamicMessage);
             Assert.IsNull(destination.Payload);
 
+            // Test merging unset source message fields with inner field paths.
+            destination = new NestedTestAllTypes
+            {
+                Payload = new TestAllTypes
+                {
+                    SingleInt32 = 1000,
+                    SingleUint32 = 2000
+                }
+            };
+            Merge(new FieldMaskTree().AddFieldPath("payload.singleint32").AddFieldPath("payload.singleuint32"),
```

Comment:
Should these be payload.single_int32 and payload.single_uint32?

If not, I'm not sure what this is meant to be testing... and if so, shouldn't they both be 0 as per the other test you've added?

Reply:
The other test has `options.ReplacePrimitiveFields = true`

Reply:
Good point, I was unfamiliar with the field paths. You're right that this is wrong then.

Reply:
I removed the test because it's only really relevant after #12748 has also been addressed, so it belongs there.